libs/stats/README.md

```diff
@@ -13,21 +12,179 @@ pip install --extra-index-url="https://packages.dea.ga.gov.au" odc-stats
 
-### Prepare tasks for processing:
+### Steps to run stats
 
-From a machine that has access to your database, run:
+### 1- Save tasks
+
+From your sandbox (or a machine that has access to your database), run:
+
+```
+odc-stats save-tasks --frequency annual --grid au-30  ga_ls8c_ard_3
+```
+
+The above command will generate the following files i.e. a csv file containing a list of tasks for all the years in the database, a cache file that will be used as an input to the stats cli, and several geojson files (one for each year).  
+
+```
+ga_ls8c_ard_3_all-2013--P1Y.geojson
+ga_ls8c_ard_3_all-2014--P1Y.geojson
+ga_ls8c_ard_3_all-2015--P1Y.geojson
+ga_ls8c_ard_3_all-2016--P1Y.geojson
+ga_ls8c_ard_3_all-2017--P1Y.geojson
+ga_ls8c_ard_3_all-2018--P1Y.geojson
+ga_ls8c_ard_3_all-2019--P1Y.geojson
+ga_ls8c_ard_3_all-2020--P1Y.geojson
+ga_ls8c_ard_3_all-2021--P1Y.geojson
+ga_ls8c_ard_3_all.csv
+ga_ls8c_ard_3_all.db
+```
+
+The csv file contains the list of tasks for all the years and consists of x, y coordinates of each tile to be processed as well as the counts of datasets and satellite observations.
+
+geojson files are useful when selecting test regions as well as for debugging specific tasks - see the example below from Landsat-8, 2015.  This example shows a tile from Australia coastal region near Sydney.
+
+<img src="odc/stats/stats/auxiliary/screenshot-L8-2015.png" alt="drawing" width="1000"/>
+
+
+### 2- Run stats
+Sample command:
+```
+odc-stats run ga_ls8c_ard_3_all.db  2015--P1Y/41/13 --threads=16 --memory-limit=60Gi --resolution=30 --config cfg.yaml --location file:///localpath/
+```  
+
+Where cfg.yaml contains the following configurations (sample from a geomedian run):
+```
+plugin: gm-ls
+plugin_config:
+  resampling: bilinear
+  bands: ["blue", "red", "green", "nir", "swir1", "swir2"]
+  rgb_bands: ["red", "green", "blue"]
+  mask_band: fmask
+  cloud_classes: ["shadow", "cloud"]
+  filters: [0, 0] 
+  aux_names:
+    smad: sdev
+    emad: edev
+    bcmad: bcdev
+    count: count                   
+product:
+  name: ga_ls8c_nbart_gm_cyear_3
+  short_name: ga_ls8c_nbart_gm_cyear_3 
+  version: 3.0.0
+  product_family: geomedian
+max_processing_time: 3600
+job_queue_max_lease: 300
+renew_safety_margin: 60
+future_poll_interval: 2
+s3_acl: public-read
+# Generic product attributes
+cog_opts:
+  zlevel: 9
+  overrides:
+    rgba:
+      compress: JPEG
+      jpeg_quality: 90
+```
+
+Note that the configurations vary between different products.  A sample plugin configuration for Australia on Landsat-8 can be found here: 
+```
+https://bitbucket.org/geoscienceaustralia/datakube-apps/src/develop/workspaces/dea-dev/processing/06_stats.yaml
+```
+
+Orchestration
+-----
+
+In order to run stats on a large volume of data, use stats orchestration tool which relies on SQS queues and Kubernetes jobs.  Before running a job, one needs to create the necessary infrastructures by running the step 1 described below.
+
+### 1- Create queues and a user 
```

Comment:
Hmm... a lot of this is DEA specific. Probably needs @whatnick or @NikitaGandhi to suggest the right way to document?